tgui/packages/tgui/interfaces/MCDependencyDebug.tsx

```diff
@@ -0,0 +1,303 @@
+      left={`${(x - 1) * 500}px`}
+      p={1}
+      top={`${y * 50}px`}
+    >
+      <Stack align="center" fill>
+        <Stack.Item>
+          <Box
+            className={classes(['ObjectComponent__Port'])}
+            textAlign="center"
+          >
+            <svg
+              style={{
+                width: '100%',
+                height: '100%',
+                position: 'absolute',
+              }}
+              viewBox="0, 0, 100, 100"
+            >
+              <circle
+                stroke={'blue'}
+                strokeDasharray={`${100 * Math.PI}`}
+                strokeDashoffset={-100 * Math.PI}
+                className={`color-stroke-blue`}
+                strokeWidth="50px"
+                cx="50"
+                cy="50"
+                r="50"
+                fillOpacity="0"
+                transform="rotate(90, 50, 50)"
+              />
+              <circle cx="50" cy="50" r="50" className={`color-fill-blue`} />
+            </svg>
+            <span ref={inputRef} className="ObjectComponent__PortPos" />
+          </Box>
+        </Stack.Item>
+        <Stack.Item grow={1}>{name}</Stack.Item>
+        <Stack.Item>
+          <Box
+            className={classes(['ObjectComponent__Port'])}
+            textAlign="center"
+          >
+            <svg
+              style={{
+                width: '100%',
+                height: '100%',
+                position: 'absolute',
+              }}
+              viewBox="0, 0, 100, 100"
+            >
+              <circle
+                stroke={'blue'}
+                strokeDasharray={`${100 * Math.PI}`}
+                strokeDashoffset={-100 * Math.PI}
+                className={`color-stroke-blue`}
+                strokeWidth="50px"
+                cx="50"
+                cy="50"
+                r="50"
+                fillOpacity="0"
+                transform="rotate(90, 50, 50)"
+              />
+              <circle cx="50" cy="50" r="50" className={`color-fill-blue`} />
+            </svg>
+            <span ref={outputRef} className="ObjectComponent__PortPos" />
+          </Box>
+        </Stack.Item>
+        <Stack.Item />
+      </Stack>
+    </Box>
+  );
+};
+
+type SubsystemRef = {
+  input?: HTMLElement;
+  output?: HTMLElement;
+};
+
+type ConnectionRef = {
+  [key: string]: SubsystemRef;
+};
+
+type SubsystemCoord = {
+  input: Coord;
+  output: Coord;
+};
+
+type ConnectionData = {
+  [key: string]: SubsystemCoord;
+};
+
+type SubsystemLayer = {
+  [subsystemName: string]: number;
+};
+
+type SubsystemMap = {
+  [subsystemName: string]: Subsystem;
+};
+
+function EvaluateSubsystemLayer(
+  subsystem: Subsystem,
+  depth: number,
+  data: SubsystemLayer,
+) {
+  data[subsystem.name] = Math.max(depth, data[subsystem.name] || 0);
+  for (let i = 0; i < subsystem.dependents.length; i++) {
+    EvaluateSubsystemLayer(subsystem.dependents[i], depth + 1, data);
+  }
+}
+
+export const MCDependencyDebug = (props, context) => {
```

Comment:
```suggestion
export function MCDependencyDebug (props){
```
same as earlier. arrow function is fine here too, but context is not needed